### DISCUSSION TITLE
Security setting to disable flashing over USB

### DISCUSSION THREAD

**Discussion by JaneX8** [mriscoc/Ender3V2S1#1393] [Ideas]
I would very much like to have a mechanism to disable firmware flashing over USB and I realize this can only be done in the printer firmware.

I'm using OctoPi which is, although only accessible in my home network, a network connected web interface that has a USB interface to the printer and does with the help of a plugin allow remote firmware flashing. From a security point of view, if somehow OctoPrint was hacked malicious firmware could be flashed remotely. While if USB flashing could be disabled on the printer itself, the risk disappears. As it would require physical access to the SD card.

The main reason I think remote flashing is such a risk is that like malicious firmware could be flashed that for example overheats the limits of the printer while not reporting the current temperatures to the display or serial port. This is an inherit fire safety risk and target for cyber attacks with potential physical damage.

With all current conflicts going on in the world, the increase in cyber attack and potentially misconfigured OctoPrint instances being exposed to the internet. This is a real risk that could be solved by simply having a setting that disables USB flashing.

**Comment by mriscoc** [comment]
You can disable USB firmware update by commenting out:
https://github.com/mriscoc/Ender3V2S1/blob/4f65c631d46ecbf4a2944716aa8c7a169e7d7432/Marlin/Configuration_adv.h#L1902-L1908

**Reply by JaneX8** [reply]
Thank you. I will try to make a custom build. However, I think having this as an option in all builds simply in the settings menu will make it more accessible for everyone.

You could even consider disabling it by default for aforementioned security reasons.
